src/commands/mobile_app/upload.rs

```diff
@@ -399,61 +406,7 @@ fn upload_file(
     } else {
         println!("Nothing to upload, all files are on the server");
     }
-
-    poll_assemble(
-        api,
-        checksum,
-        &checksums,
-        org,
-        project,
-        sha,
-        build_configuration,
-    )?;
-    Ok(())
-}
-
-fn poll_assemble(
-    api: &AuthenticatedApi,
-    checksum: Digest,
-    chunks: &[Digest],
-    org: &str,
-    project: &str,
-    sha: Option<&str>,
-    build_configuration: Option<&str>,
-) -> Result<()> {
-    debug!("Polling assemble for checksum: {}", checksum);
-
-    let progress_style = ProgressStyle::default_spinner().template("{spinner} Processing files...");
-    let pb = ProgressBar::new_spinner();
-
-    pb.enable_steady_tick(100);
-    pb.set_style(progress_style);
-
-    let response = loop {
-        let response =
-            api.assemble_mobile_app(org, project, checksum, chunks, sha, build_configuration)?;
-
-        if response.state.is_finished() {
-            break response;
-        }
-
-        std::thread::sleep(ASSEMBLE_POLL_INTERVAL);
-    };
-
-    pb.finish_with_duration("Processing");
-
-    if response.state.is_err() {
-        let message = response.detail.as_deref().unwrap_or("unknown error");
-        bail!("Failed to process uploaded files: {}", message);
-    }
-
-    if response.state.is_pending() {
-        info!("File upload complete (processing pending on server)");
-    } else {
-        info!("File processing complete");
-    }
-
-    Ok(())
+    Ok(response.artifact_id)
```

Comment:
You need to make a call to the assemble endpoint after uploading the chunks, so that the server begins the assembly process. I don't expect these changes to work, as the code is currently written.